### PR TITLE
transform WP_Error into the Mastodon API format

### PR DESCRIPTION
**Pull request by pfefferle**
See https://docs.joinmastodon.org/entities/Error/

### CODE REVIEW

File: includes/class-mastodon-api.php [hunk 1]
```diff
@@ -73,6 +74,26 @@ public function allow_cors() {
...
+	/**
+	 * Reformat error responses to match the Mastodon API.
+	 *
+	 * @see https://docs.joinmastodon.org/entities/Error/
+	 *
+	 * @param array $result The API result.
+	 *
+	 * @return array The reformatted result.
+	 */
+	public function reformat_error_response( $result ) {
```
Comment (akirk) [inline includes/class-mastodon-api.php:86]: Above you requested 3 parameters (and the hook has 3 parameters) but here you're ignoring two. I think they should be either added here or the number of parameters should be changed above.

File: includes/class-mastodon-api.php [hunk 2]
```diff
@@ -73,6 +74,26 @@ public function allow_cors() {
...
+		if ( ! empty( $result['code'] ) && ! empty( $result['message'] ) ) {
```
Comment (akirk) [inline includes/class-mastodon-api.php:87]: Is the existence of these two keys enough to ensure that this is an error? Could we also check `http_response_code()`?

**Review by akirk — CHANGES_REQUESTED** [review]
Nice that it can be done through a hook! I think we need another quick iteration.

**Review by akirk — APPROVED** [review]

**Comment by akirk** [comment]
I am having some second thoughts here, I think we also need to ensure that we are modifying this error response only for a request that was handled by the EMA plugin. Maybe do use the third parameter `WP_REST_Request $request` and simply check if the rest route starts with `api/`? What do you say?

**Comment by pfefferle** [comment]
oh, good point!

**Pull request merged**